### PR TITLE
tests: net: tx_timestamp move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move tx_timestamp tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

File: tests/net/tx_timestamp/src/main.c [hunk 1]
```diff
@@ -277,7 +277,7 @@ static void test_timestamp_setup_all(void)
 	net_if_unregister_timestamp_cb(&timestamp_cb_3);
 }
 
-static void test_timestamp_cleanup(void)
+static void test_timestamp_cleanup(void *dummy)
```
Comment (enjiamai) [inline tests/net/tx_timestamp/src/main.c:280]: Why do we add this?

File: tests/net/tx_timestamp/src/main.c [hunk 2]
```diff
@@ -486,17 +487,13 @@ static void test_check_timestamp_after_enabling(void)
 	}
 }
 
-void test_main(void)
+ZTEST(tx_timestamp_test_suite, test_tx_timestamp)
```
Comment (nashif) [inline tests/net/tx_timestamp/src/main.c:490]: no need for _test_suite here.

**Review by nashif — CHANGES_REQUESTED** [review]

File: tests/net/tx_timestamp/src/main.c (hunk 1 above)
Comment (hunterhu3000) [reply tests/net/tx_timestamp/src/main.c:280]: the tear down function needs a parameter of void *

File: tests/net/tx_timestamp/src/main.c (hunk 2 above)
Comment (hunterhu3000) [reply tests/net/tx_timestamp/src/main.c]: Updated. Thanks to review.

**Comment by hunterhu3000** [comment]
The 	test_timestamp_setup_all and test_check_timestamp_after_enabling depends on the previous functions and need to be put in one function.
CONFIG_ZTEST_SHUFFLE=y can be set in this test.

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Review by rlubos — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Comment by hunterhu3000** [comment]
@nashif Could you please help to review and +1, so that the PR can be merged in to the mainline? Thanks

**Review by nashif — APPROVED** [review]

**Review by jukkar — APPROVED** [review]

**Comment by carlescufi** [comment]
@hunterhu3000 please rebase

**Comment by nashif** [comment]
this change is not needed anymore, it was already done in main branch.

**Pull request closed without merging**